File: backend/game-service/src/main/java/com/popplar/gameservice/entity/BaseEntity.java
package com.popplar.gameservice.entity;

import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@MappedSuperclass
@Getter
@EntityListeners(AuditingEntityListener.class)
@RequiredArgsConstructor
public class BaseEntity {

    private Boolean deleted = false;

    @CreatedDate
    private LocalDateTime createdDate;

    public void setDeleted() {
        this.deleted = true;
    }

    public void setCreatedDate(LocalDateTime createdDate) {
        this.createdDate = createdDate;
    }
}

File: backend/member-service/src/main/kotlin/com/hotspot/global/entity/BaseEntity.kt
package com.hotspot.global.entity

import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@MappedSuperclass
@EntityListeners(value = [AuditingEntityListener::class])
class BaseEntity(
    @CreatedDate
    var createdAt: LocalDateTime = LocalDateTime.now(),

    @LastModifiedDate
    val updatedAt: LocalDateTime = LocalDateTime.now(),

    var deleted: Boolean = false
) {
    fun delete() {
        deleted = true
    }
}
